test/integration/scheduler/scheduler_test.go

```diff
@@ -265,58 +217,29 @@ var _ = ginkgo.Describe("Scheduler", func() {
-			createdJob2 := &batchv1.Job{}
-			gomega.Consistently(func() *bool {
-				lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
-				gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob2)).Should(gomega.Succeed())
-				return createdJob2.Spec.Suspend
-			}, framework.ConsistentDuration, framework.Interval).Should(gomega.Equal(pointer.Bool(true)))
+			wl2 := testing.MakeWorkload("on-demand-wl2", ns.Name).Queue(queue.Name).Request(corev1.ResourceCPU, "4").Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBePending(ctx, k8sClient, wl2)
 			framework.ExpectPendingWorkloadsMetric(cq, 1)
 			framework.ExpectAdmittedActiveWorkloadsMetric(cq, 1)
 
 			ginkgo.By("Third small workload starts")
-			job3 := testing.MakeJob("on-demand-job3", ns.Name).Queue(queue.Name).Request(corev1.ResourceCPU, "1").Obj()
-			gomega.Expect(k8sClient.Create(ctx, job3)).Should(gomega.Succeed())
-			createdJob3 := &batchv1.Job{}
-			gomega.Eventually(func() *bool {
-				lookupKey := types.NamespacedName{Name: job3.Name, Namespace: job3.Namespace}
-				gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob3)).Should(gomega.Succeed())
-				return createdJob3.Spec.Suspend
-			}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
-			gomega.Expect(createdJob3.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+			wl3 := testing.MakeWorkload("on-demand-wl3", ns.Name).Queue(queue.Name).Request(corev1.ResourceCPU, "1").Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl3)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, wl3)
 			framework.ExpectPendingWorkloadsMetric(cq, 1)
 			framework.ExpectAdmittedActiveWorkloadsMetric(cq, 2)
 
 			ginkgo.By("Second big workload starts after the first one is deleted")
-			gomega.Expect(k8sClient.Delete(ctx, job1, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(gomega.Succeed())
-			// Simulated deletion of workload by ownerReference of job1.
-			gomega.Expect(k8sClient.Delete(ctx,
-				&kueue.Workload{ObjectMeta: metav1.ObjectMeta{
-					Name:      job1.Name,
-					Namespace: job1.Namespace,
-				}})).Should(gomega.Succeed())
-			gomega.Eventually(func() *bool {
-				lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
-				gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob2)).Should(gomega.Succeed())
-				return createdJob2.Spec.Suspend
-			}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
-			gomega.Expect(createdJob2.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+			gomega.Expect(k8sClient.Delete(ctx, wl1, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, wl2)
```

Comment:
the previous test checked the flavor too

Reply:
flavor admission check added to all workloads of the test